relay_rpc/src/auth.rs

```diff
@@ -99,90 +90,29 @@ impl AuthToken {
     }
 }
 
-#[derive(Serialize, Deserialize)]
-pub struct JwtHeader<'a> {
-    pub typ: &'a str,
-    pub alg: &'a str,
-}
-
-impl<'a> JwtHeader<'a> {
-    pub fn is_valid(&self) -> bool {
-        self.typ == JWT_HEADER_TYP && self.alg == JWT_HEADER_ALG
-    }
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct JwtClaims<'a> {
-    pub iss: &'a str,
-    pub sub: &'a str,
-    pub aud: &'a str,
-    pub iat: i64,
-    pub exp: i64,
-}
-
-impl<'a> JwtClaims<'a> {
-    pub fn validate(
-        &self,
-        aud: &HashSet<String>,
-        time_leeway: impl Into<Option<i64>>,
-    ) -> Result<(), JwtVerificationError> {
-        let time_leeway = time_leeway
-            .into()
-            .unwrap_or(JWT_VALIDATION_TIME_LEEWAY_SECS);
-        let now = Utc::now().timestamp();
-
-        if now - time_leeway > self.exp {
-            return Err(JwtVerificationError::Expired);
-        }
-
-        if now + time_leeway < self.iat {
-            return Err(JwtVerificationError::NotYetValid);
-        }
-
-        if !aud.contains(self.aud) {
-            return Err(JwtVerificationError::InvalidAudience);
-        }
-        Ok(())
-    }
-}
-
 pub fn encode_auth_token(
     key: &Keypair,
-    sub: &str,
-    aud: &str,
+    sub: impl Into<String>,
```

Comment:
Why do we need to pass an owned (most likely) type here? Can it be `impl AsRef`?

Reply:
The use case I had was to avoid cloning a string. If you have an owned `String` that you don't need, you use it as an argument. If you don't, use a reference. This also accepts everything that can be made into a string.

So it's not really a requirement, but if you had `&str` or `impl AsRef<str>` as the argument type, you'd have to clone it even if you don't need to own the original and could avoid cloning.

Reply:
Seems like we only need to clone it because `JwtBasicClaims` requires `String`. Can we just use `&str` there or parametrize it over string type?

Reply:
It's not a problem either way, but I think a flexible signature like this solves this (minor) issue.